rudder-core/src/main/scala/com/normation/rudder/domain/nodes/Node.scala

```diff
@@ -135,22 +180,41 @@ object CompareProperties {
    *   the property
    * - if the value is the emtpy string, remove
    *   the property
+   *
+   * Each time, we have to check the provider of the update to see if it's compatible.
+   * Node that in read-write mode, the provider is the last who wrote the property.
    */
-  def updateProperties(props: Seq[NodeProperty], updates: Option[Seq[NodeProperty]]) = {
+  def updateProperties(props: Seq[NodeProperty], updates: Option[Seq[NodeProperty]], provider: NodePropertyProvider): Box[Seq[NodeProperty]] = {
+    import NodePropertyMode._
     updates match {
-      case None => props
+      case None => Full(props)
       case Some(u) =>
-        val values = u.map { case NodeProperty(k, v) => (k, v)}.toMap
+        val values = u.map { case NodeProperty(k, v, p, m) => (k, (v, p, m))}.toMap
```

Comment:
Maybe we could have done a Map[String,NodeProperty] here ?

u.map(p => (p.name,p)